interfaces/udev/backend.go

```diff
@@ -55,16 +62,43 @@ func (b *Backend) Setup(snapInfo *snap.Info, devMode bool, repo *interfaces.Repo
 	if err != nil {
 		return fmt.Errorf("cannot obtain udev security snippets for snap %q: %s", snapName, err)
 	}
-	content, err := b.combineSnippets(snapInfo, snippets)
+	combinedSnippets, err := b.combineSnippets(snapInfo, snippets)
 	if err != nil {
 		return fmt.Errorf("cannot obtain expected udev rules for snap %q: %s", snapName, err)
 	}
-	glob := fmt.Sprintf("70-%s.rules", interfaces.SecurityTagGlob(snapName))
 	dir := dirs.SnapUdevRulesDir
 	if err := os.MkdirAll(dir, 0755); err != nil {
 		return fmt.Errorf("cannot create directory for udev rules %q: %s", dir, err)
 	}
-	return ensureDirState(dir, glob, content, snapName)
+
+	if len(combinedSnippets) == 0 {
```

Comment:
What happens on removals? The original logic used ensureDirState, which deletes all files matching the glob which are not provided for creation inside the content map. The new logic seems bogus in that regard.